crux-core/src/crux/tx.clj

```diff
@@ -357,10 +365,32 @@
   (all-content-hashes [this eid]
     (idx/all-content-hashes snapshot eid))
 
+  (decode-value [this a content-hash value-buffer]
+    (if (c/can-decode-value-buffer? value-buffer)
+      (c/decode-value-buffer value-buffer)
+      (let [doc (db/get-document this content-hash)
+            value-or-values (get doc a)]
+        (cond
+          (not (idx/multiple-values? value-or-values))
+          value-or-values
+
+          (nil? value-buffer)
+          (first (idx/vectorize-value value-or-values))
+
+          :else
+          (loop [[x & xs] (idx/vectorize-value value-or-values)]
```

Comment:
OOI - is this much quicker than `(first (filter ...))`?

Reply:
Unsure, this code was already there, it was moved from `crux.query/bound-results-for-var`, but did wonder about its performance characteristics when moving it.